frontend/src/app/components/modal-content/modal-content.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, OnInit, Input, EventEmitter, Output } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+  selector: 'app-modal-content',
+  templateUrl: './modal-content.component.html',
+  styleUrls: ['./modal-content.component.sass']
+})
+export class ModalContentComponent implements OnInit {
+
+  @Input() public content;
+  @Output() passEntry: EventEmitter<boolean> = new EventEmitter();
+  constructor(public activeModal: NgbActiveModal) { }
+
+  ngOnInit(): void {
+  }
+
+  confirm(){
+    this.passEntry.emit(true);
+    this.activeModal.close();
+  }
+  decline(){
+    this.passEntry.emit(false);
+    this.activeModal.close();
```

Comment:
Could you simplify a bit? Instead of creating your own event invoke this.activeModal.dismiss() (it automatically reject) when you want to decline or cancel an operation. Then you will able to get the result in modalRef.result.
Fox example, modalRef.result.then(() => true).catch(() => false)